langchain/retrievers/azure_cognitive_search.py

```diff
@@ -52,10 +53,25 @@ def validate_environment(cls, values: Dict) -> Dict:
         )
         return values
 
-    def _build_search_url(self, query: str) -> str:
+
+    def _build_search_url(self, query: str, top_k: int) -> str:
+        """Builds the search URL for querying the search service.
+        Args:
+            query (str): The search query to be executed.
+            top_k (int): The number of search results to retrieve.
+        Returns:
+            str: The constructed search URL.
+        Raises:
+            None
+        Examples:
+            >>> client = SearchClient()
+            >>> url = client._build_search_url("example query", 10)
+            >>> print(url)
+            https://search-service.search.windows.net/indexes/index-name/docs?api-version=api-version&search=example%20query&top=10
+        """
         base_url = f"https://{self.service_name}.search.windows.net/"
         endpoint_path = f"indexes/{self.index_name}/docs?api-version={self.api_version}"
-        return base_url + endpoint_path + f"&search={query}"
+        return base_url + endpoint_path + f"&search={query}" + f"&top={top_k}"
```

Comment:
Needs to be `f"&$top={top_k}"`. Weird, I know, but ACS has called to call the param `$top`

Also, if `top_k` is optional, you need to handle that case here

Reply:
Shall I raise the PR in your branch [acs-top-n](https://github.com/UmerHA/langchain/tree/acs-top-n), which you created

Reply:
Shall I raise the PR in your branch [acs-top-n](https://github.com/UmerHA/langchain/tree/acs-top-n), which you created